src/kubernetes.js

```diff
@@ -35,4 +35,97 @@ export class KubernetesChaos {
     killNamespace(namespaceName) {
         return this.client.namespaces.delete(namespaceName)
     }
-}
\ No newline at end of file
+
+class JobRunner {
+
+    constructor(client) {
+        this.client = client
+    }
+
+   runJobOnNodes(namespace, jobName, nodes, image, command){
+        let jobs = []
+	for (let i = 0; i < nodes.length; i++) {
+          let name = `${jobName}-${(Math.random() + 1).toString(36).substring(7)}`;
+          console.log(`Creating job ${name} on node ${nodes[i]}`)
+          const job = this.client.jobs.create({
+            namespace: namespace,
+            name: name,
+            image: image,
+            command: command,
+            node_name: nodes[i]
+          })
+	  jobs.push(name)
+        }
+        return jobs
+    }
+
+    cleanJobs(namespace, jobs){
+        for (let i = 0; i < jobs.length; i++) {
+           console.log(`Deleting job ${jobs[i]}`)
+           this.client.jobs.delete(jobs[i],namespace)
+        }
+        return jobName
+    }
+
+}
+
+export class StressNodesAttack {
+    constructor(client) {
+        this.client = client
+	this.jobRunner = new JobRunner(client)
+        this.name = 'stress-nodes'
+        this.namespace = 'default'
+        this.cpu = '0'
+        this.io = '0'
+        this.vm = '0'
+        this.vmBytes = '256MB'
+        this.timeout = '10'
+    }
+
+    inNamespace(namespace) {
+        this.namespace = namespace
+        return this
+    }
+    inNodes(nodes) {
+        this.nodes = nodes
+        return this
+    }
+    withName(name) {
+        this.name = name
+        return this
+    }
+    withCPU(cpu) {
+        this.cpu = cpu
+        return this
+    }
+    withIO(io) {
+        this.io = io
+        return this
+    }
+    withVM(vm) {
+        this.vm = vm
+        return this
+    }
+    withVMBytes(vmBytes) {
+        this.vmBytes = vmBytes
+        return this
+    }
+    withTimeout(timeout) {
+        this.timeout = timeout
+        return this
+    }
+    start() {
+        this.jobs = this.jobRunner.runJobOnNodes(
+            this.namespace,
+            this.name,
+            this.nodes,
+            'progrium/stress',
+            ['stress', '--cpu', this.cpu, '--io', this.io, '--vm', this.vm, '--vm-bytes', this.vmBytes, '--timeout', this.timeout]
```

Comment:
@dgzlopes Maybe it would be necessary to set the resource limits for the pods generated by the jobs to prevent them to be evicted or throttled. Otherwise, the load they would impose in the node would be less than expected.

Reply:
Fully agree :+1: